[Feat/#288] ai 피드백 기능구현

ukongee opened the pull request:
## 변경 사항
ex) 로그인 시, 구글 소셜 로그인 기능을 추가했습니다.


## 테스트 결과 (테스트를 했으면)
ex) 베이스 브랜치에 포함되기 위한 코드는 모두 정상적으로 동작해야 합니다. 결과물에 대한 스크린샷, GIF, 혹은 라이브 데모가 가능하도록 샘플API를 첨부할 수도 있습니다.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
  - Removed outdated AI feedback handling logic to prevent errors related to missing feedback.

- **Refactor**
  - Streamlined AI feedback logic by removing unnecessary API calls and simplifying the handling process across components, resulting in a more reliable user experience.
  - Updated the `CommentForm` to delegate AI feedback handling to the parent component, enhancing modularity.
  - Adjusted the `CommentSection` to manage AI feedback requests internally, improving component independence.

- **Chores**
  - Added a new configuration file for Yarn dependency management.
  - Improved readability of the `.gitignore` file.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dlwldnjs1009 committed:
be/feat: github actions 배포 설정

BellYun committed:
fix : PositionModal 테스트 수정

BellYun committed:
fix : 주석 추가

BellYun committed:
test ci 설정

BellYun committed:
fix : any 제거

Commit:
modified:   frontend/.gitignore

Commit:
Merge branch 'develop' of https://github.com/Techeer-blabla/techeer-resume into develop

Commit:
modified:   frontend/package.json

BellYun committed:
feat : front ci 수정

BellYun committed:
fix : fix conflict

dlwldnjs1009 committed:
feat: k6부하테스트 + 모니터링 환경구축

YunJuwon0825 committed:
feat : 더미 유저 데이터 생성  API

YunJuwon0825 committed:
feat : 더미 유저 데이터 생성 완료

jung2941 committed:
refactor: WebP 파일로 이미지 포맷 변환

jung2941 committed:
feat: 북마크 조회 방식 수정

jung2941 committed:
feat: 북마크 삭제 기능 추가

YunJuwon0825 committed:
fix : 주석 풀음

YunJuwon0825 committed:
fix : PR 수정

YunJuwon0825 committed:
chore: 연결 완료

YunJuwon0825 committed:
feat: redis에  refresh token 추가하는 로직 완성

jung2941 committed:
feat: 유저 정보 관리 및 로그인 상태 업데이트 기능 추가

Gwanghyeon-k committed:
be/fix: 북마크 controller 수정

Gwanghyeon-k committed:
be/fix: 북마크 converter 수정

Gwanghyeon-k committed:
be/fix: 북마크 response 값 수정

Gwanghyeon-k committed:
be/fix: 북마크 service 수정

Gwanghyeon-k committed:
be/fix: 북마크 테스트 비활성화

Gwanghyeon-k committed:
be/fix: 북마크 service 수정

Gwanghyeon-k committed:
be/fix: 북마크 컨트롤러 수정

YunJuwon0825 committed:
feat : refresh token cache 기능 구현

jung2941 committed:
feat: 프로필 이미지 통일

jung2941 committed:
feat: 이력서 및 북마크 관리 기능 개선

jung2941 committed:
refactor: 불필요한 eslint 규칙 제거 및 axios 임포트 경로 수정

jung2941 committed:
refactor: Navbar 컴포넌트에서 프로필 이미지 관련 주석 제거

Gwanghyeon-k committed:
be/fix: 변수명 변경

YunJuwon0825 committed:
fix : 수정 완료

YunJuwon0825 committed:
fix : test 잠시 종료

YunJuwon0825 committed:
fix : test 잠시 종료2

YunJuwon0825 committed:
fix : 오류해결

jung2941 committed:
refactor: 컴포넌트 이미지 크기 조정

jung2941 committed:
feat : BookmarkTap에 지연 로딩 적용

Commit:
feat: 이력서 업로드 기능 수정

Commit:
이력서 등록 수정

Commit:
코드래빗 리뷰 반영

lsh1215 committed:
feat : 이력서 삭제 로직 구현

이력서 소프트 삭제 구현, 해당 이력서 유저만 이력서 삭제 가능

jung2941 committed:
feat: 북마크 관리 기능 추가

jung2941 committed:
feat: 날짜 포맷팅 기능 추가

jung2941 committed:
patch: DS_Store 삭제

Delete .DS_Store

BellYun committed:
feat : authStore 추가

BellYun committed:
feat : 토큰 갱신 로직 추가

BellYun committed:
refactor : 인증 로직 수정

BellYun committed:
fix : 비로그인시에도 이력서 조회 가능

Commit:
feat: 필터링 수정

Commit:
feat: 안내문구 넣기

lsh1215 committed:
fix : 사용되지 않는 코드 삭제

lsh1215 committed:
test : 이력서 소프트삭제 유닛 테스트 구현

Entity 생성을 매번 해야하는 번거로움을 없애기 위해 util 메서드 구현. 이후 이력서 소프트 삭제 유닛 테스트 코드 작성

lsh1215 committed:
fix : application.yml로부터 값 넘겨받기

기존 env에서 넘겨받는 방식에서 application.yml에서 넘겨 받도록 변경

lsh1215 committed:
fix : bean 중복 문제 해결

String, String으로 key value를 사용하고 있는 상황이여서 StringRedisTemplate으로 변경

lsh1215 committed:
feat : redis health check

redis 연결 성공 or 실패 여부 확인 가능하도록 connection check 작성

lsh1215 committed:
refactor : log 추가

lsh1215 committed:
chore : spring servlet의 multipart size 변경

upload file size 문제 해결을 위해 spring servlet multipart size 키움

lsh1215 committed:
test : 이력서 통합 테스트 로직 작성

BellYun committed:
fix : LoginStore 수정

BellYun committed:
feat : commentForm 비로그인시 사용 불가

BellYun committed:
fix : 의존성 배열

lsh1215 committed:
fix : 이력서 view count 추가

이력서 조회순으로 조회 가능하도록 view count 추가, 이력서 개별 조회 로직에서 view count 증가

lsh1215 committed:
fix : 이력서 예외 처리 로직 제거

이력서가 없을 때 404 에러를 발생시키던 것에서 빈 값을 주는 방법으로 변경

lsh1215 committed:
fix : 페이지네이션 로직 수정

findAll 제거 및 page -> slice 로 변경

lsh1215 committed:
fix : view_count 추가

view count 추가로 개별 조회 시 view가 증가, 조회 로직에서 output값으로 view_count 추가

lsh1215 committed:
fix : update로 인해 transactional 추가

lsh1215 committed:
fix : 비로그인 회원 조회 가능

Spring Security Config file을 수정하여 여러 이력서 조회를 하는 로직이 가능하도록 함

lsh1215 committed:
fix : 비로그인 회원 조회 가능

다중 조회용 API endpoint 추가

Gwanghyeon-k committed:
[be/fix] ai Controller 엔드포인트 및 dto 변환 수정

Gwanghyeon-k committed:
[be/fix] aiFeedback 생성 및 조회 수정

Gwanghyeon-k committed:
[be/fix] 모든 피드백 조회 로직 수정

Gwanghyeon-k committed:
[be/fix] ai 응답값 수정

Gwanghyeon-k committed:
[be/fix] feedbackServiceTest 임시 비활성화

ukongee committed:
feat: ai피드백 기능

ukongee committed:
feat: 최신순, 조회순 기능구현

ukongee committed:
feat: 쿼리 적용

BellYun committed:
fix : eslint 수정

BellYun committed:
fix : eslint 수정

BellYun committed:
fix : eslint  수정

ukongee committed:
ai 피드백

ukongee committed:
ai피드백 수정

dlwldnjs1009 committed:
fix: backend-cd.yml

이벤트 트리거:
on: 아래에 push 이벤트는 develop 브랜치에, 그리고 pull_request 이벤트는 main과 develop 브랜치에 대해 트리거되도록 수정

기본 작업 디렉토리:
jobs.deploy.defaults.run.working-directory를 backend로 지정해 모든 run 단계가 backend 디렉토리에서 실행되도록 수정

dlwldnjs1009 committed:
fix: backend-cd.yml

be/** 브랜치에 push되었을 때는 트리거되지 않고, 오직 develop 브랜치에 push되거나 pull request가 생성될 때만 GitHub Actions가 실행되도록 수정

dlwldnjs1009 committed:
chore: 백엔드 배포 테스트

dlwldnjs1009 committed:
fix: 백엔드 github actions 수정

지금은 PR이 열리거나 업데이트될 때도 워크플로우가 트리거되므로, 병합되기 전에도 해당 워크플로우가 실행된다.
develop 브랜치에 push했을 때만 실행되도록 수정

dlwldnjs1009 committed:
chore: 백엔드 배포 테스트

lsh1215 committed:
refactor : 주석 추가

lsh1215 committed:
test : 이력서 생성 로직 단위 테스트 작성

lsh1215 committed:
test : 이력서 생성 로직 단위 테스트 작성

lsh1215 committed:
refactor : 이력서 생성 시 파라미터로 User 객체 전달

이력서 내부 Builder 패턴 생성자를 없앰, 이력서 생성 시 파라미터로 User를 전달하여 service로직 분리

lsh1215 committed:
test : User 객체 생성용 util class 일부 수정

lsh1215 committed:
refactor : controller 일부 수정

lsh1215 committed:
refactor : 통합 -> 인수

API 두 개를 테스트하고 시나리오 기반으로 작성한 테스트 코드여서 인수 테스트로 명칭 수정

lsh1215 committed:
refactor : 이력서 생성 통합 테스트 작성(멀티쓰레딩)

lsh1215 committed:
test : 인수 테스트에서 검증한 로직 삭제

lsh1215 committed:
test : 쓰레드 한개로 해서 일단 돌아가도록 변경

lsh1215 committed:
refactor : 상수 명시

스레드 수를 담을 변수 설정을 가독성을 위해 static final을 이용해 상수로 명시

lsh1215 committed:
fix : 이력서 조회 시 인증/ 인가 제외

lsh1215 committed:
fix : 유저 이름 일부로도 이력서 조회 가능하도록

BellYun reviewed (dismissed):
깔끔하게 작성해주신것 같네요! 좋습니다!

BellYun committed:
design : 이력서 정보 컴포넌트 퍼블리싱 수정

BellYun committed:
design : 댓글 목록 컴포넌트 퍼블리싱 수정

BellYun committed:
design : 댓글 작성 컴포넌트 퍼블리싱 수정

BellYun committed:
fix : 북마크 기능 롤백

BellYun committed:
fix : any 제거

jung2941 committed:
design: 네비게이션 바 디자인 수정

BellYun committed:
test : 퍼블리싱 변경에 따른 테스트 코드 수정

BellYun committed:
fix : props 미수정 발견 및 수정

jung2941 committed:
design: PostCard 컴포넌트 스타일 개선 및 레이아웃 수정

jung2941 committed:
design: BannerCard 컴포넌트 디자인 수정

jung2941 committed:
chore: 코드에 주석 추가

jung2941 committed:
design: Modal의 위치 스타일 수정

ukongee committed:
modified:   frontend/src/components/comment/CommentForm.tsx

ukongee committed:
ai 피드백 연동

ukongee committed:
디자인 수정

BellYun reviewed (dismissed):
LGTM~

lsh1215 committed:
fix : UserInfoResponse 추가

lsh1215 committed:
fix : 기본 생성자 추가

Jackson은 역직렬화할 때 기본 생성자(No-Args Constructor)가필요한데 없어서 GlobalExceptionHandler로 빠지는 문제 해결

lsh1215 committed:
fix : 생성시간 반환 추가

jung2941 committed:
design : 이력서 업로드 페이지 퍼블리싱 수정

jung2941 committed:
feat: 이력서 삭제 기능 추가

jung2941 committed:
feat: 이력서 미리보기 구현

jung2941 committed:
fix: ProtectedRoute 주석제거

jung2941 committed:
feat: ResumeItem에 onUpdate prop 추가

BellYun committed:
chore : Sentry

BellYun committed:
feat : 커스텀 에러 설정

BellYun committed:
fix : 오탈자

BellYun committed:
fix : eslint

ukongee committed:
ai피드백 수정

BellYun reviewed (dismissed):
LGTM 💯 ~ 
후속처리까지 감사합니다~

ukongee committed:
Merge branch 'develop' of https://github.com/Techeer-blabla/techeer-resume into feat/#288

ukongee committed:
ai피드백 마지막

BellYun reviewed (approved):
수정 사항이 계속 추가되네요, 수고하셨습니다~
